src/superplaceholder.js

```diff
@@ -52,43 +58,50 @@
 			self.el.addEventListener('blur', function () {
 				self.cleanUp();
 			});
-		}
-		else {
+		} else {
 			self.processText(0);
 		}
 	};
 
 	PlaceHolder.prototype.cleanUp = function () {
 		// Stop timeouts
-		for (var i = this.timeouts.length; i--;) {
+		var i;
+
+		for (i = this.timeouts.length; i--;) {
 			clearTimeout(this.timeouts[i]);
 		}
+
 		this.el.setAttribute('placeholder', this.originalPlaceholder);
 		this.timeouts.length = 0;
 	};
 
 	PlaceHolder.prototype.typeString = function (str, callback) {
-		var self = this,
-			timeout;
+		var self = this;
+		var timeout;
+		var i;
+		var length;
 
 		if (!str) { return false; }
+
 		function setTimeoutCallback(index) {
 			// Add cursor `|` after current substring unless we are showing last
 			// character of the string.
 			self.el.setAttribute('placeholder', str.substr(0, index + 1) + (index === str.length - 1 || !self.options.showCursor ? '' : self.options.cursor));
+
 			if (index === str.length - 1) {
 				callback();
 			}
 		}
-		for (var i = 0; i < str.length; i++) {
+
+		for (i = 0, length = str.length; i < length; i++) {
```

Comment:
Caching length is unnecessary now-a-days with modern browsers doing it already internally